Mathlib/Analysis/SpecialFunctions/OrdinaryHypergeometric.lean

```diff
@@ -0,0 +1,282 @@
+# Ordinary hypergeometric function in a Banach algebra
+
+In this file, we define `ordinaryHypergeometric`, the _ordinary_ or _Gaussian_ hypergeometric
+function in a topological algebra `𝔸` over a field `𝕂` given by: $$
+_2\mathrm{F}_1(a\ b\ c : \mathbb{K}, x : \mathbb{A}) = \sum_{n=0}^{\infty}\frac{(a)_n(b)_n}{(c)_n}
+\frac{x^n}{n!}   \,,
+$$
+with $(a)_n$ is the ascending Pochhammer symbol (see `ascPochhammer`).
+
+This file contains the basic definitions over a general field `𝕂` and notation for `₂F₁`,
+as well as showing that terms of the series are zero if any of the `(a b c : 𝕂)` are sufficiently
+large non-positive integers, rendering the series finite. In this file "sufficiently large" means
+that `-n < a` for the `n`-th term, and similarly for `b` and `c`.
+
+- `ordinaryHypergeometricSeries` is the `FormalMultilinearSeries` given above for some `(a b c : 𝕂)`
+- `ordinaryHypergeometric` is the sum of the series for some `(x : 𝔸)`
+- `ordinaryHypergeometricSeries_eq_zero_of_nonpos_int` shows that the `n`-th term of the series is
+zero if any of the parameters are sufficiently large non-positive integers
+
+## `[RCLike 𝕂]`
+
+If we have `[RCLike 𝕂]`, then we show that the latter result is an iff, and hence prove that the
+radius of convergence of the series is unity if the series is infinite, or `⊤` otherwise.
+
+- `ordinaryHypergeometricSeries_eq_zero_iff` is iff variant of
+`ordinaryHypergeometricSeries_eq_zero_of_nonpos_int`
+- `ordinaryHypergeometricSeries_radius_eq_one` proves that the radius of convergence of the
+`ordinaryHypergeometricSeries` is unity under non-trivial parameters
+
+## Notation
+
+`₂F₁` is notation for `ordinaryHypergeometric`.
+
+## References
+
+See <https://en.wikipedia.org/wiki/Hypergeometric_function>.
+
+## Tags
+
+hypergeometric, gaussian, ordinary
+-/
+
+open Nat
+
+section Field
+
+variable {𝕂 : Type*} (𝔸 : Type*) [Field 𝕂] [Ring 𝔸] [Algebra 𝕂 𝔸] [TopologicalSpace 𝔸]
+  [TopologicalRing 𝔸]
+
+/-- `ordinaryHypergeometricSeries 𝔸 (a b c : 𝕂)` is a `FormalMultilinearSeries`.
+Its sum is the `ordinaryHypergeometric` map. -/
+noncomputable def ordinaryHypergeometricSeries (a b c : 𝕂) : FormalMultilinearSeries 𝕂 𝔸 𝔸 :=
+  fun n => ((n !⁻¹ : 𝕂) * (ascPochhammer 𝕂 n).eval a * (ascPochhammer 𝕂 n).eval b *
+    ((ascPochhammer 𝕂 n).eval c)⁻¹ ) • ContinuousMultilinearMap.mkPiAlgebraFin 𝕂 n 𝔸
+
+variable {𝔸} (a b c : 𝕂)
+
+/-- `ordinaryHypergeometric (a b c : 𝕂) : 𝔸 → 𝔸` is the ordinary hypergeometric map, defined as the
+sum of the `FormalMultilinearSeries` `ordinaryHypergeometricSeries 𝔸 a b c`.
+-/
+noncomputable def ordinaryHypergeometric (x : 𝔸) : 𝔸 :=
+  (ordinaryHypergeometricSeries 𝔸 a b c).sum x
+
+@[inherit_doc]
+notation "₂F₁" => ordinaryHypergeometric
+
+theorem ordinaryHypergeometricSeries_apply_eq (x : 𝔸) (n : ℕ) :
+    (ordinaryHypergeometricSeries 𝔸 a b c n fun _ => x) =
+    ((n !⁻¹ : 𝕂) * (ascPochhammer 𝕂 n).eval a * (ascPochhammer 𝕂 n).eval b *
+    ((ascPochhammer 𝕂 n).eval c)⁻¹ ) • x ^ n := by
```

Comment:
Style nit: when line-length forces us to break a term over multiple lines, it is desirable to signal this with indentation:
```suggestion
    (ordinaryHypergeometricSeries 𝔸 a b c n fun _ => x) =
      ((n !⁻¹ : 𝕂) * (ascPochhammer 𝕂 n).eval a * (ascPochhammer 𝕂 n).eval b *
        ((ascPochhammer 𝕂 n).eval c)⁻¹ ) • x ^ n := by
```